lightning/src/ln/channelmanager.rs

```diff
@@ -5682,7 +5689,22 @@ where
 				let mut events = self.pending_events.lock().unwrap();
 				events.append(&mut new_intercept_events);
 			}
-			if push_forward_event { self.push_pending_forwards_ev() }
+			if push_forward_event {
+				let mut pending_events = self.pending_events.lock().unwrap();
```

Comment:
I see `push_pending_forwards_ev()` is also used in `ChannelManager::fail_htlc_backwards_internal`, would this also be a problem there?